tests/e2e/e2e_lsm_test.go

```diff
@@ -185,19 +187,33 @@ func (s *IntegrationTestSuite) testLSM() {
 	)
 
 	// Redeem tokens for shares
-        redeemAmount := sendAmount.Sub(ibcTransferAmount)
+	redeemAmount := sendAmount.Sub(ibcTransferAmount)
 	s.executeRedeemShares(s.chainA, 0, redeemAmount.String(), validatorAAddr.String(), gaiaHomePath, fees.String())
 
 	// check redeem success
 	s.Require().Eventually(
 		func() bool {
+			balanceRes, err := getSpecificBalance(chainEndpoint, validatorAAddr.String(), shareDenom)
+			s.Require().NoError(err)
+			if !balanceRes.Amount.IsNil() && balanceRes.Amount.IsZero() {
+				return false
+			}
+
 			delegationRes, err := queryDelegation(chainEndpoint, validatorAddressA, validatorAAddr.String())
 			delegation := delegationRes.GetDelegationResponse().GetDelegation()
 			s.Require().NoError(err)
 
-			balanceRes, err := getSpecificBalance(chainEndpoint, delegatorAddress, shareDenom)
-			s.Require().NoError(err)
-			return (balanceRes.Amount.IsNil() || balanceRes.Amount.IsZero()) && delegation.Shares.GT(selfBondedShares)
+			if !delegation.Shares.Equal(selfBondedShares.Add(sdk.NewDecFromInt(redeemAmount.Amount))) {
+				return false
+			}
+
+			// // check tokenize share record module account balance
```

Comment:
is this for the rewards? I think they're accumulated passively if you redeem and then just want to check the reward owner (i.e. recipient of TransferTokenizeShareRecord)

You might want to use a different account for this instead of the validator so the staking rewards are not conflated with the validator's rewards

Reply:
This is the reward for checking liquid staked tokens (not validator rewards)